node_package/src/helpers/renderHelper.ts

```diff
@@ -0,0 +1,19 @@
+import supportsReactCreateRoot from './supportsReactCreateRoot';
+import * as legacyRootHandler from './legacyRootHandler';
+import * as modernRootHandler from './modernRootHandler';
+
+
+let toExport;
+
+if (supportsReactCreateRoot) {
+  toExport = legacyRootHandler;
+} else {
+  toExport = modernRootHandler;
+}
```

Comment:
Struggling to find a nicer way to get this going, can't think of a way to do a dynamic import and then reexport

Reply:
Would be fairly trivial with top-level await though 🤔